nomad/structs/structs.go

```diff
@@ -1741,6 +1741,9 @@ func (t *Task) Validate() error {
 	if t.Name == "" {
 		mErr.Errors = append(mErr.Errors, errors.New("Missing task name"))
 	}
+	if strings.ContainsAny(t.Name, `/\`) {
```

Comment:
Extract this into a constant and add some comments around why we are doing this.